Package dependency for immutables needs a fix

I'm using Python 3.13.0 and pip 25.0.1 on an arm64 processor.

Running `pip install -e . ` or `pip install immutables==0.20` or `pip install --no-cache-dir immutables==0.20` generates this error:

```
Collecting immutables==0.20
  Downloading immutables-0.20.tar.gz (88 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: immutables
  Building wheel for immutables (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for immutables (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [34 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/_protocols.py -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/_testutils.py -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/_version.py -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/__init__.py -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/map.py -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      running egg_info
      writing immutables.egg-info/PKG-INFO
      writing dependency_links to immutables.egg-info/dependency_links.txt
      writing requirements to immutables.egg-info/requires.txt
      writing top-level names to immutables.egg-info/top_level.txt
      reading manifest file 'immutables.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      adding license file 'LICENSE-APACHE'
      adding license file 'NOTICE'
      writing manifest file 'immutables.egg-info/SOURCES.txt'
      copying immutables/_map.pyi -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      copying immutables/py.typed -> build/lib.macosx-14.0-arm64-cpython-313/immutables
      running build_ext
      building 'immutables._map' extension
      creating build/temp.macosx-14.0-arm64-cpython-313/immutables
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -DNDEBUG=1 -I/Users/me/environments/myenv/include -I/opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13 -c immutables/_map.c -o build/temp.macosx-14.0-arm64-cpython-313/immutables/_map.o -O2 -std=c99 -fsigned-char -Wall -Wsign-compare -Wconversion
      immutables/_map.c:1287:12: error: call to undeclared function '_PyLong_Format'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       1287 |     tmp2 = _PyLong_Format(tmp1, 2);
            |            ^
      immutables/_map.c:1287:10: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
       1287 |     tmp2 = _PyLong_Format(tmp1, 2);
            |          ^ ~~~~~~~~~~~~~~~~~~~~~~~
      2 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for immutables
Failed to build immutables
ERROR: Failed to build installable wheels for some pyproject.toml based projects (immutables)
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -17,7 +17,7 @@ dependencies = [
     "safetensors~=0.5.2",
     "accelerate~=1.3.0",
     "pydantic~=2.10.6",
-    "immutables==0.20",
+    "immutables==0.21",
     "transformers>=4.45.2",
     "tokenizers>=0.20.1",
     "huggingface_hub",

